### PR TITLE
add `search` function to `Data.List.Relation.Unary.All`

### LINKED ISSUE #989: searching in list
Searching in a list can be expressed via lib-1.2 as
```  
search  :  ∀ {p} {P : Pred A p} → Decidable P → (xs : List A) →
                                                (∃ (\x → P x)) ⊎ (¬ Any P xs)
search P? xs  with any P? xs
...    | yes anyP-xs =   inj₁ (satisfied anyP-xs)
...    | no ¬anyP-xs =   inj₂ ¬anyP-xs 
```
First, is there a ready function in lib-1.2 to express this?
Second, it is more desirable to have (for a setoid)
```
search  :  ∀ {p} {P : Pred A p} → Decidable P → (xs : List A) →
                                               (∃ (\x → x ∈ xs × P x)) ⊎ (¬ Any P xs)
```
Has  lib-1.2 something for  this?

### PR DESCRIPTION
**Pull request by jamesmckinna**
Fixes #989 

NB. UPDATED `CHANGELOG` added
* In `Data.List.Relation.Unary.All`:
  ```agda
  search : Decidable P → ∀ xs → All (∁ P) xs ⊎ Any P xs
  ```

### CODE REVIEW

**Comment by JacquesCarette** [comment]
I'd approve if there were a CHANGELOG.

**Comment by jamesmckinna** [comment]
> I'd approve if there were a CHANGELOG.

Currently all the v2.2-badged PRs I'm posting (or reviewing ;-)) are without `CHANGELOG`, against the inevitable reset after the eventual v2.1 release...

**Comment by mildsunrise** [comment]
maybe it would ease some burden for everyone if we started special branches for every release, i.e. `v2.1`, and we could keep working on `master` for the next minor (or even major) release?

File: src/Data/List/Relation/Unary/All.agda [hunk 1]
```diff
@@ -232,6 +232,9 @@ decide p∪q (x ∷ xs) with p∪q x
 ... | inj₂ qx = inj₂ (here qx)
 ... | inj₁ px = Sum.map (px ∷_) there (decide p∪q xs)
 
+search : Decidable P → ∀ xs → All (∁ P) xs ⊎ Any P xs
+search P? = decide (Sum.swap ∘ (Dec.toSum ∘ P?))
```
Comment (Taneb) [inline src/Data/List/Relation/Unary/All.agda:236]: Are the inner brackets necessary here? `∘` should be associative

File: src/Data/List/Relation/Unary/All.agda [hunk 2]
```diff
@@ -14,7 +14,7 @@ open import Data.List.Membership.Propositional renaming (_∈_ to _∈ₚ_)
 import Data.List.Membership.Setoid as SetoidMembership
 open import Data.Product.Base as Product
   using (∃; -,_; _×_; _,_; proj₁; proj₂; uncurry)
-open import Data.Sum.Base as Sum using (inj₁; inj₂)
+open import Data.Sum.Base as Sum using (_⊎_;inj₁; inj₂)
```
Comment (Taneb) [inline src/Data/List/Relation/Unary/All.agda:17]:
```suggestion
open import Data.Sum.Base as Sum using (_⊎_; inj₁; inj₂)
```

**Review by Taneb — CHANGES_REQUESTED** [review]

**Comment by jamesmckinna** [comment]
> maybe it would ease some burden for everyone if we started special branches for every release, i.e. `v2.1`, and we could keep working on `master` for the next minor (or even major) release?

That sounds good, so shoutout to the release-Meister @MatthewDaggitt for comment!? (The actual machinery of producing a release is still a bit mysterious to me...)

**Review by Taneb — APPROVED** [review]
Approved pending changelog

**Comment by MatthewDaggitt** [comment]
Yup happy to create a release branch. We've never really had the quantity of contributors around a release time that made it a necessity before.

**Comment by jamesmckinna** [comment]
> Yup happy to create a release branch. We've never really had the quantity of contributors around a release time that made it a necessity before.

But let's capitalise on the momentum that our new (and old!) contributors are bringing to the development process!?

**Comment by jamesmckinna** [comment]
*contra* the approvals (for which all thanks), I'm wondering if a sharper lemma should be provable, namely that for `Decidable P`, we can exhibit a *view* of any given list `xs` as either `All (not P) xs` or `All (not P) ys` and `Any P zs` for a *prefix* decomposition `ys ++ zs` of `xs` (while trying to prove the equivalence of various definitions of `deleteFirst` against #988 ...)

**Comment by jamesmckinna** [comment]
`CHANGELOG` entry added on the blank `v2.2-dev` template inherited from #2439 ... so a successful merge of that PR should see a more sensible diff/patch for this one...

**Review by MatthewDaggitt — APPROVED** [review]

**Pull request merged**